pyEpiabm/pyEpiabm/core/cell.py

```diff
@@ -70,16 +71,24 @@ def add_microcells(self, n):
         for _ in range(n):
             self.microcells.append(Microcell(self))
 
-    def set_id(self, id: float):
+    def set_id(self, id: str):
         """Updates ID of cell (i.e. for input from file).
 
         Parameters
         ----------
-        id : float
+        id : str
             Identity of cell
 
         """
-        self.id = id
+        # Ensure id is a string
+        if not isinstance(id, str):
+            raise TypeError("id must be of type string")
+
+        # May want to set upper limit on the number of digits
+        if re.match("^\\d+$", id):
+            self.id = id
+        else:
+            raise ValueError("id must take the correct form")
```

Comment:
This isn't very informative - maybe use an f-string to quote the ID provided and then say what the correct form is (all digits)